### PR TITLE
app-misc/gpick: gpick-0.3 will not build without wrapping lua header files in extern "C"

### PR DESCRIPTION
**Pull request by BlackCrypto103**
Signed-off-by: Blackcrypto <cryptoanon101@protonmail.com>

Patched the gpick-0.3 build

### CODE REVIEW

File: app-misc/gpick/gpick-0.3.ebuild [hunk 1]
```diff
@@ -26,9 +26,14 @@ RDEPEND="
 
 DEPEND="${RDEPEND}"
 
-S="${WORKDIR}/gpick-gpick-${PV}"
```
Comment (BlackCrypto103) [inline app-misc/gpick/gpick-0.3.ebuild:29]: This is no longer needed as ${S} directory is now gpick-0.3

File: app-misc/gpick/gpick-0.3.ebuild [hunk 2]
```diff
@@ -26,9 +26,14 @@ RDEPEND="
...
+PATCHES=( ${FILESDIR}/gpick-libc.patch )
 
 src_prepare() {
+	#echo 0.3 > .version || die
```
Comment (BlackCrypto103) [inline app-misc/gpick/gpick-0.3.ebuild]: Oops... I forgot to delete lines 32-35. Ignore these

File: app-misc/gpick/files/.version [hunk 3]
```diff
@@ -0,0 +1,4 @@
+0.3
```
Comment (BlackCrypto103) [inline app-misc/gpick/files/.version:1]: This version file is now needed for building without git

File: app-misc/gpick/gpick-0.3.ebuild [hunk 4]
```diff
@@ -8,7 +8,7 @@ inherit cmake xdg
 DESCRIPTION="Advanced color picker written in C++ using GTK+ toolkit"
 HOMEPAGE="https://github.com/thezbyg/gpick"
 SRC_URI="https://github.com/thezbyg/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
-
+RESTRICT="mirror"
```
Comment (BlackCrypto103) [inline app-misc/gpick/gpick-0.3.ebuild]: Forgot to delete this as well.  I used it during testing. I'll remove it

**Comment by Universebenzene** [comment]
@github-actions[bot] 看起来 ${FILESDIR} 是得用双引号引起来的

File: app-misc/gpick/gpick-0.3.ebuild (hunk 1 above)
Comment (Universebenzene) [reply app-misc/gpick/gpick-0.3.ebuild:29]: `${FILESDIR}` QA报告要求用引号引起来

File: app-misc/gpick/gpick-0.3.ebuild [hunk 5]
```diff
@@ -26,9 +25,10 @@ RDEPEND="
 
 DEPEND="${RDEPEND}"
 
-S="${WORKDIR}/gpick-gpick-${PV}"
+PATCHES=( ${FILESDIR}/gpick-libc.patch )
 
 src_prepare() {
+	cp ${FILESDIR}/.version .
```
Comment (Universebenzene) [inline app-misc/gpick/gpick-0.3.ebuild]: `${FILESDIR}` QA报告要求用引号引起来。另外这句漏 `|| die` 了

File: app-misc/gpick/gpick-0.3.ebuild [hunk 6]
```diff
@@ -41,15 +41,6 @@ src_configure() {
...
-	local so
-	for so in *.so
-	do
-		doexe "${so}"
-	done
-	popd || die
-
 	cmake_src_install
 
 	mv "${D}/usr/share/doc/${PN}" "${D}/usr/share/doc/${P}" || die
```
Comment (Universebenzene) [inline app-misc/gpick/gpick-0.3.ebuild]: doc的目录是不是应该用 `${PF}` 而不只是 `${P}` ？

Comment (BlackCrypto103) [reply app-misc/gpick/gpick-0.3.ebuild]: Yes. I'll change this

**Review by Universebenzene — APPROVED** [review]

**Pull request merged**